### DISCUSSION TITLE
how set_cache_size affects Read/Write performance and cache mechanisms

### DISCUSSION THREAD

**Discussion by dongbin86** [vincent-herlemont/native_db#267] [General]
1. How does `set_cache_size` impact read and write performance?
2. Are the read and write caches managed separately?
3. Are these caches directly allocated by the process, or are they part of the system's page cache?
4. If the cache is directly allocated by the process, will there be issues at startup if it exceeds available memory?
5. Does the read cache operate on an eviction mechanism?
6. Does the write cache act as a buffer that flushes based on time or size?

**Comment by vincent-herlemont** [comment]
@dongbin86 I think you will find most of your answers in [this comment](https://github.com/vincent-herlemont/native_db/discussions/240#discussioncomment-10853141), if you still have questions don't hesitate to ask them again.
